### PR TITLE
Move packages on DBus

### PR DESCRIPTION
**Pull request by poncovka**
Drop the support for the `%packages` section from UI. Use DBus API
to work with the selection of packages, groups and environments.

**TODO:**

- [x] Run kickstart tests.
- [x] Fix the kdump addon.
- [x] Fix the oscap addon.

**Blocked by:** Christmas

### CODE REVIEW

File: pyanaconda/ui/tui/spokes/software_selection.py [hunk 1]
```diff
@@ -54,47 +54,40 @@ def __init__(self, data, storage, payload):
...
-                    self._selected_environment = environments[0]
+            # Apply the initial selection.
+            self.apply()
 
-        # Apply the initial selection
-        self._apply()
+        # Check the initial software selection.
+        self.execute()
 
         # Wait for the software selection thread that might be started by _apply().
```
Comment (rvykydal) [inline pyanaconda/ui/tui/spokes/software_selection.py:92]: _apply -> apply ?

**Review by rvykydal — APPROVED** [review]
Looks good to me. As far as I can tell.

File: pyanaconda/ui/tui/spokes/software_selection.py [hunk 2]
```diff
@@ -281,60 +281,32 @@ def ready(self):
...
-                for addon_id in available_addons:
-                    if addon_id in self.addons:
-                        self.payload.select_group(addon_id)
+        # Clear groups data.
+        self._selection.excluded_groups = []
+        self._selection.groups_package_types = {}
 
-                changed = True
+        # Select valid groups.
+        # FIXME: Remove invalid groups from selected groups.
```
Comment (jkonecny12) [inline pyanaconda/ui/tui/spokes/software_selection.py:296]: Couldn't this be a big pain for users?

**Review by jkonecny12 — APPROVED** [review]
Other than the below. Looks good to me!

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/ui/tui/spokes/software_selection.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/ui/tui/spokes/software_selection.py]: Fixed.

**Comment by poncovka** [comment]
Rebased.

File: pyanaconda/ui/tui/spokes/software_selection.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/ui/tui/spokes/software_selection.py:296]: GUI works this way. When you choose some groups and then change the environment, the invalid groups will break the transaction. You want to keep groups that are valid for the selected environment.

**Pull request merged**